Set final day button admin front-end.

kseniaxu0704 opened the pull request:
This PR is to create a set-up final day button for the admin system - Other Links → User Management for front end 

<img width="1428" alt="image" src="https://user-images.githubusercontent.com/108507783/201501650-d30d71e6-f4b6-4c8d-a37a-a046d9a01719.png">
<img width="1439" alt="image" src="https://user-images.githubusercontent.com/108507783/201501654-c5965815-d4c8-4c2a-9691-eab798c490b8.png">

It is possible to set a final day for each user in our database and delete the final day if mind change.

Note : if you want to pull the code and test the effect, please also pull the PR Yanxu/admin/finaldaybutton for backend. Only they both change will show the final effect

EvianTan commented on src/actions/userManagement.js:
```diff
@@ -152,3 +152,25 @@ export const updateUserFinalDayStatus = (user, status, finalDayDate) => {
   };
 };
+
+export const updateUserFinalDayStatusu = (user, status, finalDayDate,isSet) => {
```
The function name is not clear meaningful, what does the last char "u" mean here? And you already have one function `updateUserFinalDayStatus`,  consider changing this one to a more meaningful one.

kseniaxu0704 replied:
Sure, this one will revise in the revision.

EvianTan commented on src/actions/userManagement.js:
```diff
@@ -152,3 +152,25 @@ export const updateUserFinalDayStatus = (user, status, finalDayDate) => {
+  const userProfile = Object.assign({}, user);
+  userProfile.endDate = finalDayDate;
+  userProfile.isActive = status === "Active";
+  userProfile.isSet = isSet ==="FinalDay";
```
Consider changing all `isSet` to `isFinalDaySet` to be more accurate. We might have more properties of `userProfile` to accord set or not set in the future.

kseniaxu0704 replied:
`isSet` for now not only be modified in this pr, but have already existed in the repo, I may need to create a new Sprint task for it to track the modification tasks. Since this change is only name change and does not have effect to this commit, will balance to new tasks.

EvianTan replied:
Understood.

EvianTan commented on src/languages/en/ui.js:
```diff
@@ -48,4 +48,5 @@ export const ACTIVE_TEAMS = 'Active Teams';
-export const SET_FINAL_DAY = 'Set Final Day';
\ No newline at end of file
+export const SET_FINAL_DAY = 'Set Final Day';
+export const Management = 'Manage Final Day';
```
It's better to use all caps here for consistency. Change it to `export const MANAGE_FINAL_DAY = 'Manage Final Day'`. Don't forget change all the places it is used.

EvianTan commented on src/languages/en/ui.js:
```diff
@@ -32,7 +32,7 @@ export const LAST_NAME = 'Last Name';
 export const EMAIL = 'Email';
 export const JOB_TITLE = 'Job Title';
-export const WEEKLY_COMMITTED_HRS = 'Weekly Committed Hrs';
+export const WEEKLY_COMMITTED_HRS = 'Wkly Committed Hrs';
```
Revert this change. you can see the left side and the right side of the equation are almost same, except for capitalization.

EvianTan commented on src/utils/enums.js:
```diff
@@ -1,11 +1,16 @@
+import { unset } from "lodash";
```
Remove this one, It's not used in this file. I guess you mis-import it by line 34...

EvianTan commented on src/components/UserManagement/UserTableHeader.jsx:
```diff
@@ -37,7 +38,11 @@ const UserTableHeader = React.memo((props) => {
       </th>
       <th scope="col" id="usermanagement_pause" id="user_pause">
         {' '}
-        {PAUSE + ' / ' + RESUME}
```
Remove Line 10 since we don't need `RESUME` here.

EvianTan commented:
Hi @kseniaxu0704 , terrific job on those two PRs!!

✅ Verified changes:
1. Add set/delete final day column on user management page.
2. Set/delete final day function works well
3. user profile's end date changed related to the set/delete function

💡 Things can be improved:
1. Consider combining `updateUserFinalDayStatus` and `updateUserFinalDayStatusu` functions as one, with `isSet` as the default/optional paramenter
<img width="566" alt="image" src="https://user-images.githubusercontent.com/5071040/202578205-f066b4ac-bf50-46ae-9ed5-7cdc96ae4bed.png">
You can see those two functions just have minors different.


2. Still the formatting issues
With Prettier installed on VScode, you can use this command `prettier "the\file\path" --write` to help fix all the formatting problems in one file. That's a very convenient tool. 
Below image is an example for running `prettier ".\src\actions\userManagement.js" --write`
<img width="557" alt="image" src="https://user-images.githubusercontent.com/5071040/202578871-6a8a9e3c-cde3-4f8e-ad81-36500b650511.png">
Please run this command for all files you changed in this PR

kseniaxu0704 commented:
> Hi @kseniaxu0704 , terrific job on those two PRs!!
> 
> ✅ Verified changes:
> 
> 1. Add set/delete final day column on user management page.
> 2. Set/delete final day function works well
> 3. user profile's end date changed related to the set/delete function
> 
> 💡 Things can be improved:
> 
> 1. Consider combining `updateUserFinalDayStatus` and `updateUserFinalDayStatusu` functions as one, with `isSet` as the default/optional paramenter
> 
> <img alt="image" width="566" src="https://user-images.githubusercontent.com/5071040/202578205-f066b4ac-bf50-46ae-9ed5-7cdc96ae4bed.png">
> 
> You can see those two functions just have minors different.
> 2. Still the formatting issues
>    With Prettier installed on VScode, you can use this command `prettier "the\file\path" --write` to help fix all the formatting problems in one file. That's a very convenient tool.
>    Below image is an example for running `prettier ".\src\actions\userManagement.js" --write`
> 
> <img alt="image" width="557" src="https://user-images.githubusercontent.com/5071040/202578871-6a8a9e3c-cde3-4f8e-ad81-36500b650511.png">
> 
> Please run this command for all files you changed in this PR

Hi Yiyun. Thanks for reviewing my PR. I changed the coding style problem using the prettier one you recommended. It is REALLY helpful !! Appreciate! 

As for the two similar problems, The first function is used in the set final day button in the user account - view profile page, which already exists in our main. And the second function is used on the user management page for the admin account. When I write the first function, in order to accordingly to our HighestGoodNetWork previous coding, I use react hook. For this time, for the second function, I found the author for the user-management page using React life cycle to create that. Therefore for consistency, I also used that. This is why I rewrote the second function.

It may be better to set it as a new PR since it will involve a lot of other code. It is really good advice. I will combine the two functions in the future.

LeeSingChanYau commented on src/languages/en/ui.js:
```diff
@@ -32,7 +32,7 @@ export const LAST_NAME = 'Last Name';
 export const EMAIL = 'Email';
 export const JOB_TITLE = 'Job Title';
-export const WEEKLY_COMMITTED_HRS = 'Weekly Committed Hrs';
+export const WKLY_COMMITTED_HRS = 'Wkly Committed Hrs';
```
Try to be consistent when naming variables, not a big issue as of right now. Your variables seemed to be mostly spelled right, but keep in mind to spell variables right.

kseniaxu0704 replied:
I named weekly committed hours based on the requirement from Jae. After abbreviation， it possible to  look better in one line.

kseniaxu0704 replied:
I guess you mean the WKLY_COMMITTED_HRS, I use abbreviation based on the requirement from Jae. So they all it on one line.

kseniaxu0704 commented on src/actions/userManagement.js:
```diff
@@ -15,13 +15,13 @@ import moment from 'moment';
 export const getAllUserProfile = () => {
   const userProfilesPromise = axios.get(ENDPOINTS.USER_PROFILES);
-  return async (dispatch) => {
+  return async dispatch => {
```
Yes, I use Prettier extension to formate every file I change.
